rest/api.go

```diff
@@ -346,7 +348,19 @@ func (h *handler) handlePprofProfile() error {
 }
 
 func (h *handler) handlePprofBlock() error {
+	sec, err := strconv.ParseInt(h.rq.FormValue("seconds"), 10, 64)
+	if sec <= 0 || err != nil {
+		sec = 30
+	}
+	if h.server.blockRunning.IsTrue() {
+		return base.HTTPErrorf(http.StatusForbidden, "Can only run one block profile at a time")
+	}
+	h.server.blockRunning.Set(true)
+	runtime.SetBlockProfileRate(1)
+	defer runtime.SetBlockProfileRate(0)
```

Comment:
This should be reset prior to unlocking at line 363

Reply:
Done